sql/src/main/java/org/apache/druid/sql/calcite/planner/DruidSqlValidator.java

```diff
@@ -348,6 +398,42 @@ private DatasourceTable validateInsertTarget(
     }
   }
 
+  /**
+   * Clustering is a kind of ordering. We push the CLUSTERED BY clause into the source query as
+   * an ORDER BY clause. In an ideal world, clustering would be outside of SELECT: it is an operation
+   * applied after the data is selected. For now, we push it into the SELECT, then MSQ pulls it back
+   * out. This is unfortunate as errors will be attributed to ORDER BY, which the user did not
+   * actually specify (it is an error to do so.) However, with the current hybrid structure, it is
+   * not possible to add the ORDER by later: doing so requires access to the order by namespace
+   * which is not visible to subclasses.
+   */
+  private void rewriteClusteringToOrderBy(SqlNode source, DruidSqlIngest ingestNode, @Nullable SqlNodeList catalogClustering)
```

Comment:
instead of this creative reflectioned approach ; how about overriding the `createSelectNamespace`;
in that method you will see:
* the current `SqlSelect` node
* the encosingNode - which will be `SqlInsert` or something...so can give access the catalog orderby somehow
* full control on hiw the `SelectNamespace` is being created - it might be even possible to supply an alternate implementation

Reply:
Thank you for the suggestion, this worked great! Let me know if good now.

Reply:
much better - thank you for the updates!